src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -99,20 +115,37 @@ public VisitResult(TypeWithState resultType, TypeSymbolWithAnnotations lValueTyp
         /// </summary>
         private readonly ArrayBuilder<(BoundReturnStatement, TypeSymbolWithAnnotations)> _returnTypesOpt;
 
-        /// <summary>
-        /// An optional callback for callers to receive notification of the inferred type and nullability
-        /// of each expression in the method. Since the walker may require multiple passes, the callback
-        /// may be invoked multiple times for a single expression, potentially with different nullability
-        /// each time. The last call for each expression will include the final inferred type and nullability.
-        /// </summary>
-        private readonly Action<BoundExpression, TypeSymbolWithAnnotations> _callbackOpt;
-
         /// <summary>
         /// Invalid type, used only to catch Visit methods that do not set
         /// _result.Type. See VisitExpressionWithoutStackGuard.
         /// </summary>
         private static readonly TypeWithState _invalidType = new TypeWithState(ErrorTypeSymbol.UnknownResultType, NullableFlowState.NotNull);
 
+        /// <summary>
+        /// Contains the map of expressions to inferred nullabilities and types used by the optional rewriter phase of the
+        /// compiler.
+        /// </summary>
+        private readonly PooledDictionary<BoundExpression, (NullabilityInfo Info, TypeSymbol Type)> _analyzedNullabilityMapOpt;
```

Comment:
>PooledDictionary [](start = 25, length = 16)

`Dictionary<,>`

Reply:
I'm not understanding what you're asking with these.

---
In reply to: [265686131](https://github.com/dotnet/roslyn/pull/34017#discussion_r265686131) [](ancestors = 265686131)

Reply:
`NullableWalker` shouldn't care whether the dictionary is pooled or not.

---
In reply to: [265694691](https://github.com/dotnet/roslyn/pull/34017#discussion_r265694691) [](ancestors = 265694691,265686131)